src/drawing/SplashScene.h

```diff
@@ -40,13 +40,13 @@ class SplashScene : public Scene
   public:
     explicit SplashScene(sol::state state,
                          AnimationPlayerType animationPlayer,
-                         std::shared_ptr<TextureLoaderType> textureLoader,
-                         std::shared_ptr<FontLoaderType> fontLoader,
-                         std::string_view script)
+                         TextureLoaderType* textureLoader,
```

Comment:
Would change this, because in the far future you may need to close or destroy TextureLoader not at the end of program lifetime.